File: app/src/main/java/com/nikola/jakshic/truesight/model/SearchHistory.java
package com.nikola.jakshic.truesight.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;

@Entity(indices = {@Index(value = "query", unique = true)})
public class SearchHistory {

    @PrimaryKey(autoGenerate = true)
    private int id;
    @ColumnInfo(collate = ColumnInfo.NOCASE)
    private String query;

    public SearchHistory(String query) {
        this.query = query;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getQuery() {
        return query;
    }
}

File: app/src/main/java/com/nikola/jakshic/dagger/vo/SearchHistory.kt
package com.nikola.jakshic.dagger.vo

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.Index
import android.arch.persistence.room.PrimaryKey

@Entity(tableName = "search_history", indices = [Index(value = ["query"], unique = true)])
data class SearchHistory(
        @ColumnInfo(collate = ColumnInfo.NOCASE) var query: String) {
    @PrimaryKey(autoGenerate = true) var id: Long = 0
}